src/searchbar.cpp

```diff
@@ -154,6 +154,11 @@ void SearchBar::updateCompletion()
 {
     mp_typingTimer->stop();
     clearSuggestions();
+    // workaround for #541
+    // return if settings open
+    if (KiwixApp::instance()->getSettingsManager()->isSettingsViewdisplayed()) {
+        return;
+    }
     auto currentWidget = KiwixApp::instance()->getTabWidget()->currentWebView();
```

Comment:
This would be better to fix `currentWebView` implementation.

As you can see https://github.com/kiwix/kiwix-desktop/blob/master/src/tabbar.h#L27-L35, 
currentWidget return a nullptr if the current webview is the setting view (before casting it), but currentWebView don't.

And while you are here, there are few place where we use `currentWebView` method without checking the returned pointer. We should also fix this.

Reply:
Fixed as you suggested. However, I only found one place where lacking nullptr safeguard.(there are some place that don't matter if the pointer is nullptr)